### PR TITLE
Use `:not(#\20)` stackable root - more specificity

### PR DESCRIPTION
**Pull request by iamstarkov**
Fix https://github.com/MadLittleMods/postcss-increase-specificity/issues/8

Here I am with a PR 😄 

Use `:not(#\20)` stackable root to get more specificity.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,25 +24,25 @@ module.exports = postcss.plugin('postcss-increase-specifity', function(options)
...
-				// `:root:root:root .foo`
+				// `:not(#\\20):not(#\\20):not(#\\20) .foo`
 				return opts.stackableRoot.repeat(opts.repeat) + ' ' + selector;
 			});
 
 			if(opts.overrideIds) {
 				if(
 					// If an id is in there somewhere
-					(/#/).test(rule.selector) ||
+					(/#(?!\\)/).test(rule.selector) ||
```
Comment (iamstarkov) [inline index.js:45]: this thing was needed to be changed, because otherwise original regexp catches `#` in `:not(#\20)` and doesnt add `!important`

**Comment by iamstarkov** [comment]
so, @MadLittleMods what do you think?

**Comment by iamstarkov** [comment]
@MadLittleMods is something wrong?

File: package.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
   },
   "main": "index.js",
   "scripts": {
-    "test": "mocha"
+    "test": "mocha",
+    "tdd": "mocha --watch"
```
Comment (MadLittleMods) [inline package.json:20]: We could use `npm test -- --watch` instead

**Comment by MadLittleMods** [comment]
Sorry for the delay @iamstarkov 😢 

Reviewing now

File: index.js [hunk 3]
```diff
@@ -21,25 +24,25 @@ module.exports = postcss.plugin('postcss-increase-specifity', function(options)
 		css.walkRules(function(rule) {
 			rule.selectors = rule.selectors.map(function(selector) {
 				// Apply it to the selector itself if the selector is a `root` level component
-				// `html:root:root:root`
+				// `html:not(#\\20):not(#\\20):not(#\\20)`
 				if(
 					selector === 'html' ||
-					selector === ':root' ||
+					selector === ':not(#\\20)' ||
```
Comment (MadLittleMods) [inline index.js]: We should put this back to `:root`. `:not(#\\20)` would be covered by the last case `selector === opts.stackableRoot`

Comment (MadLittleMods) [inline index.js:30]: We should put this back to `:root` as it is root level thing . `:not(#\\20)` would be covered by the last case `selector === opts.stackableRoot`

**Review by MadLittleMods — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
 
+# Upcoming release
+
+ - switch from `:root` to `:not(#\20)` to get more specificity in selectors
+   - Thanks to [@subzey][] for the [idea][] and [@iamstarkov][] for the contribution [#9][]
+
+[idea]: https://twitter.com/subzey/status/829050478721896448
+[@subzey]: https://github.com/subzey
+[@iamstarkov]: https://github.com/iamstarkov
+[#9]: https://github.com/MadLittleMods/postcss-increase-specificity/pull/9
```
Comment (MadLittleMods) [inline CHANGELOG.md:11]: I prefer inline linking vs references. Seems fine to keep it for now as it will be changed in the release.

**Comment by MadLittleMods** [comment]
@iamstarkov I'll take care of the remaining points (no need for a further fix commit).

**Comment by MadLittleMods** [comment]
Merged via https://github.com/MadLittleMods/postcss-increase-specificity/pull/10

Thanks @iamstarkov 💓 😺 

Appreciate the ping again to take a look. This was sitting in my inbox for too long.

**Comment by MadLittleMods** [comment]
Published in [`postcss-increase-specificity@0.4.0`](https://github.com/MadLittleMods/postcss-increase-specificity/blob/master/CHANGELOG.md#v040---2017-5-18), https://www.npmjs.com/package/postcss-increase-specificity

**Comment by iamstarkov** [comment]
@MadLittleMods thank you

**Pull request closed without merging**